Test failure LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Framework, expectFallbackForwarder: True)

Run: [runtime-libraries-coreclr outerloop 20220512.4](https://dev.azure.com/dnceng/public/_build/results?buildId=1767026&view=ms.vss-test-web.build-test-results-tab&runId=47520410&paneView=debug&resultId=100007)

Failed test:
```
net7.0-windows-Release-x64-CoreCLR_release-Windows.11.Amd64.ClientPre.Open
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Framework, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Standard, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Core, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Net5, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Framework, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Standard, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Core, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Net5, expectFallbackForwarder: False)

net7.0-windows-Release-x64-CoreCLR_release-Windows.Amd64.Server2022.Open
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Framework, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Standard, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Core, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Net5, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Framework, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Standard, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Core, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Net5, expectFallbackForwarder: False)

net7.0-windows-Release-x64-CoreCLR_release-Windows.10.Amd64.Server2022.ES.Open
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Framework, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Standard, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Core, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Net5, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Framework, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Standard, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Core, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Net5, expectFallbackForwarder: False)

net7.0-windows-Release-x86-CoreCLR_release-Windows.7.Amd64.Open
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Framework, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Standard, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Core, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Net5, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Framework, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Standard, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Core, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Net5, expectFallbackForwarder: False)

net7.0-windows-Release-x86-CoreCLR_release-Windows.11.Amd64.ClientPre.Open
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Framework, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Standard, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Core, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Net5, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Framework, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Standard, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Core, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Net5, expectFallbackForwarder: False)

net7.0-Linux-Release-x64-CoreCLR_release-(Fedora.34.Amd64.Open)ubuntu.1604.amd64.open@mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-34-helix-20220331150839-4f64125
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Framework, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Standard, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Core, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Net5, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Framework, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Standard, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Core, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Net5, expectFallbackForwarder: False)

net7.0-Linux-Release-x64-CoreCLR_release-SLES.15.Amd64.Open
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Framework, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Standard, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Core, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Net5, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Framework, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Standard, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Core, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Net5, expectFallbackForwarder: False)

net7.0-Linux-Release-arm64-CoreCLR_release-(Ubuntu.1804.ArmArch.Open)Ubuntu.1804.ArmArch.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm64v8-20220427172132-97d8652
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Framework, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Standard, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Core, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Net5, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Framework, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Standard, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Core, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Net5, expectFallbackForwarder: False)

net7.0-Linux-Release-arm64-CoreCLR_release-(Ubuntu.2110.Arm64.Open)Ubuntu.1804.ArmArch.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-21.10-helix-arm64v8-20211116135000-0f8d97e
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Framework, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Standard, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Core, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Net5, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Framework, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Standard, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Core, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Net5, expectFallbackForwarder: False)

net7.0-OSX-Release-x64-CoreCLR_release-OSX.1200.Amd64.Open
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Framework, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Standard, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Core, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Net5, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Framework, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Standard, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Core, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Net5, expectFallbackForwarder: False)

net7.0-OSX-Release-arm64-CoreCLR_release-OSX.1200.ARM64.Open
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Framework, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Standard, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Core, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Net5, expectFallbackForwarder: True)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Framework, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Standard, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Core, expectFallbackForwarder: False)
- LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(source: \"\\r\\nusing System.Runtime.InteropServices;\\r\\nusing\"..., targetFramework: Net5, expectFallbackForwarder: False)
```
**Error message:**
```
Assert.All() Failure: 1 out of 2 items in the collection did not pass.
[0]: Item: (3,38): error CS0234: The type or namespace name 'Marshalling' does not exist in the namespace 'System.Runtime.InteropServices' (are you missing an assembly reference?)
Xunit.Sdk.SubsetException: Assert.Subset() Failure
Expected: HashSet<String> ["CS8019"]
Actual:   HashSet<String> ["CS0234"]
at Xunit.Assert.Subset[T](ISet1 expectedSuperset, ISet1 actual) in //src/xunit.assert/Asserts/SetAsserts.cs:line 71
at LibraryImportGenerator.UnitTests.TestUtils.<>c__DisplayClass4_0.<AssertPostSourceGeneratorCompilation>b__0(Diagnostic diag) in //src/libraries/System.Runtime.InteropServices/tests/LibraryImportGenerator.UnitTests/TestUtils.cs:line 115
at Xunit.Assert.<>c__DisplayClass11_01.<All>b__0(T item, Int32 index) in /_/src/xunit.assert/Asserts/CollectionAsserts.cs:line 34 at Xunit.Assert.All[T](IEnumerable1 collection, Action`2 action) in /_/src/xunit.assert/Asserts/CollectionAsserts.cs:line 61


Stack trace
   at LibraryImportGenerator.UnitTests.TestUtils.AssertPostSourceGeneratorCompilation(Compilation comp, String[] additionalAllowedDiagnostics) in /_/src/libraries/System.Runtime.InteropServices/tests/LibraryImportGenerator.UnitTests/TestUtils.cs:line 113
   at LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder(String source, TestTargetFramework targetFramework, Boolean expectFallbackForwarder) in /_/src/libraries/System.Runtime.InteropServices/tests/LibraryImportGenerator.UnitTests/Compiles.cs:line 350
--- End of stack trace from previous location ---
```
No kusto information.

--- a/src/libraries/System.Runtime.InteropServices/tests/LibraryImportGenerator.UnitTests/CodeSnippets.cs
+++ b/src/libraries/System.Runtime.InteropServices/tests/LibraryImportGenerator.UnitTests/CodeSnippets.cs
@@ -337,6 +337,8 @@ partial class Test
 
         public static readonly string DisableRuntimeMarshalling = "[assembly:System.Runtime.CompilerServices.DisableRuntimeMarshalling]";
 
+        public static readonly string UsingSystemRuntimeInteropServicesMarshalling = "using System.Runtime.InteropServices.Marshalling;";
+
         /// <summary>
         /// Declaration with parameters with <see cref="StringMarshalling"/> set.
         /// </summary>
@@ -396,7 +398,6 @@ struct Native
         /// </summary>
         public static string BasicParametersAndModifiers(string typeName, string preDeclaration = "") => $@"
 using System.Runtime.InteropServices;
-using System.Runtime.InteropServices.Marshalling;
 {preDeclaration}
 partial class Test
 {{
@@ -742,7 +743,7 @@ struct Native
                 + NonBlittableUserDefinedType()
                 + NativeTypeOut;
 
-            public static string ParametersAndModifiers = BasicParametersAndModifiers("S")
+            public static string ParametersAndModifiers = BasicParametersAndModifiers("S", UsingSystemRuntimeInteropServicesMarshalling)
                 + NonBlittableUserDefinedType(defineNativeMarshalling: true)
                 + NativeTypeRef;
 
@@ -778,7 +779,7 @@ struct Native
     public S ToManaged() => new S { b = i != 0 };
 }
 ";
-            public static string OptionalStackallocParametersAndModifiers = BasicParametersAndModifiers("S")
+            public static string OptionalStackallocParametersAndModifiers = BasicParametersAndModifiers("S", UsingSystemRuntimeInteropServicesMarshalling)
                 + NonBlittableUserDefinedType() + @"
 [CustomTypeMarshaller(typeof(S), Features = CustomTypeMarshallerFeatures.CallerAllocatedBuffer, BufferSize = 1)]
 struct Native
@@ -809,7 +810,7 @@ struct Native
     public void FromNativeValue(int value) { }
 }
 ";
-            public static string TwoStageParametersAndModifiers = BasicParametersAndModifiers("S")
+            public static string TwoStageParametersAndModifiers = BasicParametersAndModifiers("S", UsingSystemRuntimeInteropServicesMarshalling)
                 + NonBlittableUserDefinedType() + @"
 [CustomTypeMarshaller(typeof(S), Features = CustomTypeMarshallerFeatures.TwoStageMarshalling)]
 struct Native
@@ -822,7 +823,7 @@ struct Native
     public void FromNativeValue(int value) { }
 }
 ";
-            public static string TwoStageRefReturn = BasicParametersAndModifiers("S")
+            public static string TwoStageRefReturn = BasicParametersAndModifiers("S", UsingSystemRuntimeInteropServicesMarshalling)
                 + NonBlittableUserDefinedType() + @"
 [CustomTypeMarshaller(typeof(S), Direction = CustomTypeMarshallerDirection.In, Features = CustomTypeMarshallerFeatures.TwoStageMarshalling)]
 unsafe struct Native
@@ -832,7 +833,7 @@ unsafe struct Native
     public ref byte ToNativeValue() => throw null;
 }
 ";
-            public static string PinnableParametersAndModifiers = BasicParametersAndModifiers("S") + @"
+            public static string PinnableParametersAndModifiers = BasicParametersAndModifiers("S", UsingSystemRuntimeInteropServicesMarshalling) + @"
 [NativeMarshalling(typeof(Native))]
 class S
 {

